### PR TITLE
Fix: #207 - Validation for missing resolvers at Root level 

### PR DESCRIPTION
**Pull request by Nirali-Ahir**
added the validate_schema function checks whether the schema string contains a "type Query" declaration and at least one resolver (indicated by the presence of "@http"). If not, it returns an error, effectively preventing the server from starting.

/claim #207
#207

### CODE REVIEW

File: src/http/server.rs [hunk 1]
```diff
@@ -56,8 +71,12 @@ fn create_allowed_headers(headers: &HeaderMap, allowed: &HashSet<String>) -> Hea
 
   new_headers
 }
+
 pub async fn start_server(file_path: &String) -> Result<()> {
   let server_sdl = fs::read_to_string(file_path)?;
+    
+  validate_schema(&server_sdl).context("Schema validation failed")?;
```
Comment (tusharmath) [inline src/http/server.rs:78]: Validation needs to be implemented inside of `from_config.rs`

Comment (Nirali-Ahir) [reply src/http/server.rs]: aaded checks in from_config.rs

**Comment by tusharmath** [comment]
Can you run `./lint.sh` once, to fix the build failure.

**Comment by Nirali-Ahir** [comment]
> Can you run `./lint.sh` once, to fix the build failure.

done

**Comment by tusharmath** [comment]
I updated the main issue, I think that should simplify a lot of things.

**Comment by Nirali-Ahir** [comment]
> I updated the main issue, I think that should simplify a lot of things.

added validations to only required fields

File: src/blueprint/from_config.rs [hunk 2]
```diff
@@ -47,20 +47,71 @@ fn to_directive(const_directive: ConstDirective) -> Valid<Directive> {
...
+                    Ok(())
+                } else {
+                    Valid::fail("No resolver has been found in the schema".to_owned()).trace(name)
+                }
+            })
+            .trace(query_type_name)?;
+    }
+
+    // Validate mutation
+    {
```
Comment (tusharmath) [inline src/blueprint/from_config.rs:81]: use `validate_or` to combine the query and mutation checks together instead of using `?`

Comment (Nirali-Ahir) [reply src/blueprint/from_config.rs]: done

**Comment by tusharmath** [comment]
can you please run `./lint.sh`, commit and push code.

**Comment by Nirali-Ahir** [comment]
> can you please run `./lint.sh`, commit and push code.

type of required changed since last commit so update code

**Comment by tusharmath** [comment]
@Nirali-Ahir, thanks for the effort you put into this PR! 🙌 We're going to move ahead with #362, but your input really got us thinking. 🤔 Your involvement is super appreciated, and we're looking forward to seeing more from you on tailcall. 🚀 Hope to catch more of your contributions in the upcoming discussions and PRs! 💪

**Pull request closed without merging**